metrics_utility/management/validation.py

```diff
@@ -98,6 +129,169 @@ def handle_crc_ship_target():
+    Validates the list of optional collectors specified in the
+    METRICS_UTILITY_OPTIONAL_COLLECTORS environment variable against a set
+    of valid collectors.
+
+    If any invalid collectors are found, an error message is appended to the
+    provided errors list.
+
+    Args:
+        errors (list): A list to which error messages will be appended if
+            invalid collectors are found.
+
+    Environment Variables:
+        METRICS_UTILITY_OPTIONAL_COLLECTORS (str, optional): Comma-separated
+            list of collector names. Defaults to 'main_jobevent' if not set.
+
+    Notes:
+        - The set of valid collectors is defined by the global variable
+          VALID_COLLECTORS.
+        - Error messages include the invalid collector names and the list of
+          valid values.
+    """
+    collectors = os.environ.get('METRICS_UTILITY_OPTIONAL_COLLECTORS', 'main_jobevent').split(',')
+    if collectors:
+        invalid = set(collectors) - VALID_COLLECTORS
+        if invalid:
+            errors.append(f'Invalid METRICS_UTILITY_OPTIONAL_COLLECTORS: {", ".join(invalid)}. Valid values: {", ".join(VALID_COLLECTORS)}')
+
+
+def validate_ship_target(errors):
+    """
+    Validates the 'METRICS_UTILITY_SHIP_TARGET' environment variable against a set of valid ship targets.
+
+    If the environment variable is set and its value is not in the list of valid ship targets,
+    an error message is appended to the provided errors list.
+
+    Args:
+        errors (list): A list to which error messages will be appended if validation fails.
+
+    Returns:
+        str or None: The value of the 'METRICS_UTILITY_SHIP_TARGET' environment variable if set, otherwise None.
+
+    Notes:
+        - The set of valid ship targets is defined by the global variable VALID_SHIP_TARGET.
+        - Error messages include the invalid ship target and the list of valid values.
+    """
+    ship_target = os.getenv('METRICS_UTILITY_SHIP_TARGET', None)
+    if ship_target and ship_target not in VALID_SHIP_TARGET:
+        errors.append(f'Invalid METRICS_UTILITY_SHIP_TARGET: {ship_target}. Valid values: {", ".join(VALID_SHIP_TARGET)}')
+    return ship_target
+
+
+def validate_ship_path(errors, ship_target):
+    """
+    Validates the ship path environment variable based on the ship target.
+
+    Args:
+        errors (list): A list to which error messages will be appended if validation fails.
+        ship_target (str): The value of the METRICS_UTILITY_SHIP_TARGET environment variable.
+
+    Notes:
+        - For 'directory' ship target, checks if METRICS_UTILITY_SHIP_PATH is an existing directory.
+        - Appends an error message to 'errors' if the directory does not exist.
+    """
+    ship_path = os.getenv('METRICS_UTILITY_SHIP_PATH', None)
+    if ship_target == 'directory' and ship_target:
+        if not os.path.isdir(ship_path):
+            errors.append(f'Invalid METRICS_UTILITY_SHIP_PATH: {ship_path} is not an existing directory.')
```

Comment:
I'm pretty sure we have functionality somewhere to create the dir if it doesn't exist. Of course that requires that the path provided to the new dir is a valid path, and doesn't link to a file somewhere in the middle of the path.

IE, I'm not sure we want this to fail at the very beginning because somewhere in our execution we create a dir if needed.